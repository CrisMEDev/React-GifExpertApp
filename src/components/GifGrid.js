import React, { useState, useEffect } from 'react'

import { GifGridItem } from './GifGridItem';
import { getGifs } from '../helpers/getGifs';

const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);
    
    useEffect( () => {

        getGifs( category ).then( gifs => setImages( gifs ) );

    }, [category]);


    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
                {
                images.map( (gif) => <GifGridItem key={ gif.id } { ...gif } /> )
                }
            </div>
        </>
    )
}

export default GifGrid
