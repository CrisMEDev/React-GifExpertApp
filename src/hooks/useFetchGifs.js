import { useState, useEffect } from 'react';

import { getGifs } from '../helpers/getGifs';

const useFetchGifs = ( category ) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect( () => {

        getGifs( category )
            .then( imgs => setState({
                data: imgs,
                loading: false
            }));

    }, [ category ]);   // El category en el array indica que el effect se disapara cuando
                        // la categoria cambie

    return state;

}


export default useFetchGifs;
