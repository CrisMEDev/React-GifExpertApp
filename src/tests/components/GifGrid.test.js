import React from 'react';
import { shallow } from 'enzyme';

import GifGrid from '../../components/GifGrid';
import useFetchGifs from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el componente <GifGrid />', () => {
    
    test('Debe de mostrar el componente correctamente', () => {
        
        // Se define un mock para useFetchGifs que simula la implementacion de la funcion
        useFetchGifs.mockImplementation(() => {
            return {
                data: [],
                loading: true
            };
        });

        const wrapper = shallow( <GifGrid category={'warframe'} /> );
        
        expect( wrapper ).toMatchSnapshot();
        
    });
    
    test('Debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
        
        const gifs = [
            {
                id: 'ascnascknasc',
                title: 'My awesome title',
                url: 'https://the_best_url.jpg'
            },
            {
                id: 'ascascasc',
                title: 'My awesome title 2',
                url: 'https://the_best_url2.jpg'
            }
        ];
        
        // Se define un mock para useFetchGifs que simula la implementacion de la funcion
        useFetchGifs.mockImplementation(() => {
            return {
                data: gifs,
                loading: false
            };
        });

        const wrapper = shallow( <GifGrid category={'warframe'} /> );

        expect( wrapper ).toMatchSnapshot();

        // Asegurarse de que como hay data, el parrafo de 'cargando...' no se renderiza
        expect( wrapper.find('p').exists() ).toBe( false );

        // Evaluar si el componente GifGridItem existe el numero gifs de veces
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
         
    });
    
    
});


