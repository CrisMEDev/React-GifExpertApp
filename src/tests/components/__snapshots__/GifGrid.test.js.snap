// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3>
    warframe
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el componente <GifGrid /> Debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3>
    warframe
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ascnascknasc"
      key="ascnascknasc"
      title="My awesome title"
      url="https://the_best_url.jpg"
    />
    <GifGridItem
      id="ascascasc"
      key="ascascasc"
      title="My awesome title 2"
      url="https://the_best_url2.jpg"
    />
  </div>
</Fragment>
`;
